packages/peer-exchange/src/waku_peer_exchange.ts

```diff
@@ -47,7 +40,9 @@ export class WakuPeerExchange implements IPeerExchange {
    * @param components - libp2p components
    */
   constructor(public components: PeerExchangeComponents) {
-    this.multicodec = PeerExchangeCodec;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore: Libp2p does not expose internal components anymore.
+    super(PeerExchangeCodec, components);
```

Comment:
Not super nice :(

Reply:
We can avoid it by not extending `WakuPeerExchange` with `BaseProtocol` and creating a private instance of it inside of the class. That way I prefer more to be honest.

Reply:
Sounds good!

Reply:
Found a way :)